prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -104,13 +103,15 @@ class SendFunction(
                 val nextRetryItems = mutableListOf<String>()
                 val externalFileName = Report.formExternalFilename(header, workflowEngine.reportService)
                 val sentReportId = UUID.randomUUID() // each sent report gets its own UUID
-                if (receiver.transport == null) {
+                val retryItems = retryToken?.items
+                val transport = getTransport(receiver.transportType)
+                if (transport == null) {
```

Comment:
I think this logic isn't reachable anymore? Sorry my brain is kinda going in loops here. But wouldn't `receiver.transportType` set that value to `NullTransportType` which wouldn't evaluate to `null`?

Reply:
So, that's for the case of a transport type is added (which is something that is on the receiver) but the SendFunction isn't updated with a transport to support it (I think it's weird that they're not tied together well, but since they are very separate, that seemed like a possibility and I wrote a test for that case)